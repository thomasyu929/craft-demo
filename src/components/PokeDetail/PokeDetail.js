import React, { useEffect, useState } from 'react'
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import GoogleMapReact from 'google-map-react';
import { locations } from '../../assets/mock-location.json'

import './PokeDetail.scss';
import { setCheckList } from '../../state/pokeAction';

const Marker = () => <div className="marker"><i className="fas fa-map-marker-alt"></i></div>;

function PokeDetail({ pokes, checkList, setCheckList }) {
  const [poke, setPoke] = useState(null);

  const defaultProps = {
    center: { lat: 32.734778, lng: -117.152630 },
    zoom: 10
  };

  useEffect(() => {
    const path = window.location.pathname;
    const name = path.substring(path.lastIndexOf('/') + 1);

    setPoke(() => pokes.filter(poke => poke.name === name)[0]);
  }, [pokes])

  const handleCheck = (e) => {
    let bag = localStorage.getItem('bag') ? JSON.parse(localStorage.getItem('bag')) : [];

    if (e.target.checked) {
      bag.push(poke);
      console.log([...checkList, poke.name])
      setCheckList([...checkList, poke.name])
    } 
    else {
      bag = bag.filter(p => p.name !== poke.name);
      setCheckList(checkList.filter(p => p !== poke.name))
    }
    localStorage.setItem('bag', JSON.stringify(bag))
  }

  return poke && (
    <div className="details-container">
      <Link to='/'><button>Go Back</button></Link>
      <div className="info-contianer">
        <div className="details">
          <div className="basic-info">
            <img alt="poke" src={ poke.sprites.front_default }></img>
            { poke.name }
          </div>
          <div className="other-info">
            <div>Height: { poke.height }</div>
            <div>Weight: { poke.weight }</div>
            <div className="inBag"><input type="checkbox" onChange={handleCheck} defaultChecked={checkList.find(p => p === poke.name)}></input> In Bag</div>
            <div className="types">
              {
                poke.types && poke.types.map(obj => <div>{obj.type.name}</div>)
              }
            </div>
          </div>
        </div>
        <div className="map-container">
          <div id="map">
            <GoogleMapReact
              bootstrapURLKeys={{ key: 'AIzaSyD4Snui6ZviMKwOjbin7ikNjIv5x64IZ30' }}
              defaultZoom={defaultProps.zoom}
              defaultCenter={defaultProps.center}
            >
              {
                locations.map(val => val.split(',')).map((loc, i) => <Marker key={i} lat={loc[0]} lng={loc[1]}></Marker>)
              }
            </GoogleMapReact>
          </div>
        </div>
      </div>
      <div className="ability-container">
        <h2>Abilities</h2>
        {
          poke.abilities && poke.abilities.map(obj => <div><Link to="">{obj.ability.name}</Link></div>)
        }
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    pokes: state.pokeList,
    checkList: state.checkList
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setCheckList: (name) => dispatch(setCheckList(name)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(PokeDetail);
