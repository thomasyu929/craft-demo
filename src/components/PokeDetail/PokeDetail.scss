.details-container {
  display: flex;
  flex-direction: column;
  align-items: center;

  .info-contianer {
    display: flex;
    // align-items: center;

    .details {
      display: flex;
      flex-direction: column;
      min-width: 400px;
      align-items: flex-start;

      .basic-info {
        padding: 5px 0px;
        margin: 10px 0px;
        display: flex;
        flex-direction: column;
        text-align: center;
        text-transform: capitalize;
        font-weight: 700;
        font-size: 20px;
        color: rgb(49, 49, 49);

        img {
          margin-bottom: 15px;
          width: 180px;
          height: 180px;
          border: 1px solid lightgray;
          border-radius: 8px;
          box-shadow: 2px 2px 6px lightgray;
        }
      }

      .other-info {
        font-size: 15px;
        color: rgb(83, 83, 83);
        font-weight: 700;

        .inBag {
          padding: 5px 0px;

          input {
            margin: 0px 5px 0 0
          }
        }

        .types {
          display: flex;

          div {
            padding: 5px 10px 5px 0px;
            text-transform: capitalize;
            color: darkred;
          }
        }
      }
    }

    .map-container {
      display: flex;
      padding: 5px 0px;
      margin: 10px 0px;

      #map {
        width: 400px;
        height: 300px;
        background-color: bisque;
      }
    }
  }
}

.marker {
  i {
    font-size: 16px;
    color: rgb(241, 50, 50);
  }
}