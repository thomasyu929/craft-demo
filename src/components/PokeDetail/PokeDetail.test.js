import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store';
import PokeDetail from './PokeDetail';

const mockStore = configureStore([]);

describe('<PokeList.test />', () => {

  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });
 
    component = renderer.create(
      <Provider store={store}>
        <PokeDetail />
      </Provider>
    );
  });

  it('render', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
