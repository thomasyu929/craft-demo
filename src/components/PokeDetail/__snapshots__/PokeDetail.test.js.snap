// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokeList.test /> render 1`] = `null`;
